joomla/joomla-cms
# Fix placeholder showing "Select a User" when in readonly mode (e.g. modified_by)

### Summary of Changes

Fix placeholder showing "Select a User" when in readonly mode (e.g. modified_by)

**Additional changes:**
**Layout (Mootools Version ~ hathor):**
- Change htmlspecialchars($VARNAME, ENT_COMPAT, 'UTF-8') to $this->escape($VARNAME)
- Refactor code to remove the ternary overuse
- Change JHtml::script call to JHtml::_('script' ...) to allow overrides via custom register method
- Correct variable name ($exclude) in doc block to match extracted name ($excluded)
- Remove obsolete extracted variables that are not present for this field type

**Layout (jQuery / Bootstrap Version ~ isis):**
- Same as for the Mootools implementation + removed the class suffix from the hidden field
- Remove (Fix) obsolete whitespace in class attribute for the visible field when no suffix was entered

**JFormFieldUser:**
- Satisfy Code Sniffer (e.g. @since)
- Remove else block after variable initialisation
- Fixe typo in doc block of the added custom fields handling method
- Changed getExcluded method return value to match doc block
### Testing Instructions
- Open a new banner'' item in the back-end and switch to the "Publishing" tab.
  Before the patch, the "Modified By" field shows a "Select a User" placeholder. Since you cannot really select a user here, that text is obsolete in that context.

''You could also open a new article but in the hathor template, the modified_by field won't show up until the article has actually been modified which would then of course override the placeholder. Hence, a banner item.
- Apply the patch, refresh the view or open another banner item. The placeholder should be gone.

Now, this PR changes both the "old" Mootools and the jQuery / Bootstrap implementation of the layout(s), where Mootools version is currently the "default". I put default in quotes because we think of isis as the default admin template and that template has got an override for the field layout which uses jQuery / Bootstrap code to open the modal window.

So, to fully verifiy that everything works as expected, please make sure that you test the field both in hathor and isis. That is, not only confirm that the placeholder text disappears from the "Modified By" field, but also that the "Created By" still works as it should (because it is the same field).
### Caveat

I would go so far and move a few more lines, but before that, I would like your opinion first.
- Even though JHtml::script makes sure that a script only gets included ones, I would move the call for the modal stuff inside the !readonly block. The "Modified By" field is almost always(?!) added with the unset filter and readonly attribute, so no modal code is really needed here and neither is the hidden input field.

Long story short: Any objections?
